api/controller.py

```diff
@@ -143,10 +143,22 @@ def __init__(self, _db, lanes=None, testing=False):
                 sys.exit()
 
         self.testing = testing
+        self.site_configuration_last_update = (
+            Configuration.site_configuration_last_update(self._db, timeout=0)
+        )
         self.lane_descriptions = lanes
         self.setup_one_time_controllers()
         self.load_settings()
         
+    def reload_settings_if_changed(self):
+        """If the site configuration has been updated, reload the
+        CirculationManager's configuration from the database.
+        """
+        last_update = Configuration.site_configuration_last_update(self._db, timeout=0)
```

Comment:
I figured when we are first initializing the CirculationManager, we'd want to go to the database no matter what. It shouldn't make a difference, though, since I don't see a case where you would call `site_configuration_last_update` before initializing the CM.